docs/Customization/Custom-Core.rst

```diff
@@ -0,0 +1,404 @@
+.. _custom_core:
+
+Adding a custom core
+====================
+
+You may want to add a custom RISC-V core to Chipyard generator. If the top module of your core is not in Chisel, 
+you will first need to create a Verilog blackbox for it. See :ref:`incorporating-verilog-blocks` for instructions.
+Once you have a top module in Chisel, you are ready to create integrate it with Chipyard. 
+
+.. note:: 
+
+    RoCC is not supported by custom core currently. Please use Rocket or Boom if you need to use RoCC.
+
+Parameter Case Classes
+----------------------
+
+Chipyard will generate a core for every ``TileParams`` object it discovered in the current config.
+``TileParams`` is a trait containing the information needed to create a tile, and every custom core must implement
+their own version of ``TileParams``, as well as ``CoreParams`` which is passed as a field in ``TileParams``.
+
+``TileParams`` holds the parameters that are the same for every generated core, while ``CoreParams`` contains those
+that can vary from cores to cores. They must be implemented as case classes with fields that can be overridden by 
```

Comment:
I would rephrase this. TileParams holds parameters that are relevant for the Tile, while CoreParams holds parameters that are relevant for the core.

Reply:
Done.